package NYT.NChunkClient.NProto;

////////////////////////////////////////////////////////////////////////////////

message TChunkReaderStatistics
{
    required int64 data_bytes_read_from_disk = 1;
    optional int64 data_io_requests = 11;
    required int64 data_bytes_transmitted = 7 [default = 0];
    required int64 data_bytes_read_from_cache = 2;
    required int64 meta_bytes_read_from_disk = 3;
    optional int64 meta_io_requests = 14;
    optional int64 omitted_suspicious_node_count = 10;
    optional int64 data_wait_time = 4 [default = 0];
    optional int64 meta_wait_time = 5 [default = 0];
    optional int64 meta_read_from_disk_time = 9 [default = 0];
    optional int64 pick_peer_wait_time = 6 [default = 0];
    optional int64 p2p_activation_count = 12;
    optional int64 remote_cpu_time = 13;

    reserved 8;
}

////////////////////////////////////////////////////////////////////////////////
